File: SCT2/trunk/test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/EntryExitSelfTransitionTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for EntryExitSelfTransition
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class EntryExitSelfTransitionTest extends AbstractExecutionFlowTest {
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("EntryExitSelfTransition.sct");
		initInterpreter(flow);
	}
	@Test
	public void SelfTransitionToChildState() throws Exception {
		interpreter.enter();
		interpreter.runCycle();
		assertTrue(getInteger("entries") == 1);
		assertTrue(isActive("B"));
		setInteger("entries", 0);
		raiseEvent("e");
		interpreter.runCycle();
		assertTrue(getInteger("entries") == 1);
		assertTrue(getInteger("exits") == 1);
		assertTrue(isActive("C"));
	}
	@Test
	public void SelfTransitionFromChildState() throws Exception {
		interpreter.enter();
		interpreter.runCycle();
		assertTrue(getInteger("entries") == 1);
		setInteger("entries", 0);
		raiseEvent("e1");
		interpreter.runCycle();
		assertTrue(getInteger("entries") == 0);
		assertTrue(getInteger("exits") == 0);
		assertTrue(isActive("C"));
		raiseEvent("e1");
		interpreter.runCycle();
		assertTrue(isActive("B"));
		assertTrue(getInteger("entries") == 1);
		assertTrue(getInteger("exits") == 1);
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/EntryExitSelfTransitionTest/EntryExitSelfTransitionTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "EntryExitSelfTransition.h"

EntryExitSelfTransition handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		entryExitSelfTransition_init(&handle);
	}
};

void init(){
	entryExitSelfTransition_enter(&handle);
	entryExitSelfTransition_runCycle(&handle);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_entries(&handle)== 1l);
	EXPECT_TRUE(entryExitSelfTransition_isStateActive(&handle, EntryExitSelfTransition_main_region_A__region0_B));
	entryExitSelfTransitionIface_set_entries(&handle,0l);
}

TEST_F(StatemachineTest, SelfTransitionToChildState) {					
	init();
	entryExitSelfTransitionIface_raise_e(&handle);
	entryExitSelfTransition_runCycle(&handle);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_entries(&handle)== 1l);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_exits(&handle)== 1l);
	EXPECT_TRUE(entryExitSelfTransition_isStateActive(&handle, EntryExitSelfTransition_main_region_A__region0_C));
}
TEST_F(StatemachineTest, SelfTransitionFromChildState) {					
	init();
	entryExitSelfTransitionIface_raise_e1(&handle);
	entryExitSelfTransition_runCycle(&handle);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_entries(&handle)== 0l);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_exits(&handle)== 0l);
	EXPECT_TRUE(entryExitSelfTransition_isStateActive(&handle, EntryExitSelfTransition_main_region_A__region0_C));
	entryExitSelfTransitionIface_raise_e1(&handle);
	entryExitSelfTransition_runCycle(&handle);
	EXPECT_TRUE(entryExitSelfTransition_isStateActive(&handle, EntryExitSelfTransition_main_region_A__region0_B));
	EXPECT_TRUE(entryExitSelfTransitionIface_get_entries(&handle)== 1l);
	EXPECT_TRUE(entryExitSelfTransitionIface_get_exits(&handle)== 1l);
}


